From 0fb8b664a9a49b299ac3f7851df8cd9b9121e445 Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Mon, 21 Jan 2019 20:27:36 +0100
Subject: [PATCH xkeyboard-config] Map evdev keycode KEY_ROTATE_LOCK_TOGGLE to
 XF86RotationLockToggle

Add XF86RotationLockToggle, to be used as mapping for evdev's
KEY_ROTATE_LOCK_TOGGLE.

I've a Point of View P1006W-232 Windows tablet which actually has a
rotate-lock toggle-button. The latest kernel correctly generates
KEY_ROTATE_LOCK_TOGGLE events for this. So now I'm hooking up support for
it through all the higher layers.

The patch for adding the RotationLockToggle keysym to xproto is here:
https://patchwork.freedesktop.org/patch/279365/

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
---
 keycodes/evdev | 1 +
 symbols/inet   | 1 +
 2 files changed, 2 insertions(+)

diff --git a/keycodes/evdev b/keycodes/evdev
index dd2b865..5dcc9f6 100644
--- a/keycodes/evdev
+++ b/keycodes/evdev
@@ -286,6 +286,7 @@ default xkb_keycodes "evdev" {
 
 	<I372> = 372;   // #define KEY_FAVORITES           364
 	<I382> = 382;   // #define KEY_KEYBOARD            374
+	<I569> = 569;   // #define KEY_ROTATE_LOCK_TOGGLE  561
 
 	// Fake keycodes for virtual keys
 	<LVL3> =   92;
diff --git a/symbols/inet b/symbols/inet
index b97ccf2..6c9dd87 100644
--- a/symbols/inet
+++ b/symbols/inet
@@ -220,6 +220,7 @@ xkb_symbols "evdev" {
 
     key <I372>  {       [ XF86Favorites          ]       };
     key <I382>  {       [ XF86Keyboard           ]       };
+    key <I569>  {       [ XF86RotationLockToggle ]       };
 
     key <FK13>   {      [ XF86Tools         ]       };
     key <FK14>   {      [ XF86Launch5       ]       };
-- 
2.21.0

